arxlive_spotlight_annotation/src/node_modules/util.mjs

```diff
@@ -0,0 +1,53 @@
+import * as _ from 'lamb'
+import { join } from '@svizzle/utils'
+import { logger } from 'logging/logging.mjs'
+
+/* String Functions */
+
+export const toLowerString = v => v.toString().toLowerCase()
+
+// tag function to dedent template literals
+export const dedent = _.pipe([
+    v => v[0], // first argument is strings
+    _.splitBy('\n'),
+    _.mapWith(v => v.trim()),
+    _.joinWith('\n'),
+    v => v.trim()
+])
+
+export const concatenateStringifyObjectValues= _.pipe([
+    _.values,
+    _.mapWith(toLowerString),
+    join
+])
+
+/* Spotlight */
+
+export const uniquesByStringifiedObjectValues =
+    _.uniquesBy(concatenateStringifyObjectValues)
+
+/* Promises */
+
+const logErrors = v => {
+	if (v.status === 'rejected') {
+		logger.error(v.reason);
+	}
+	return v;
+};
+const removeErrors = v => v.status !== 'rejected';
+
+// TODO found out why this doesn't work (I think it has to do with await)
+// const promisesHandler = _.pipe([
+// 	async v => await Promise.allSettled(v),
+// 	_.mapWith(logErrors),
+// 	_.filterWith(removeErrors),
+// ]);
```

Comment:
leftover?

Reply:
I was hoping to ask you about this. Was trying to implement the function above using Lamb, but it became really confusing once I needed to begin using `async/await` API...

Reply:
I don't think `lamb` would supports async functions, let's keep your solution, we'll discuss options internally